packages/shared_preferences/shared_preferences/example/integration_test/shared_preferences_test.dart

```diff
@@ -102,5 +104,39 @@ void main() {
       // The last write should win.
       expect(preferences.getInt('int'), writeCount);
     });
+
+    testWidgets(
+        'string clash with lists, big integers and doubles (Android only)',
+        (WidgetTester _) async {
+      await preferences.clear();
+      // special prefixes plus a string value
+      expect(
+          // prefix for lists
+          preferences.setString(
+              'String',
+              'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBhIGxpc3Qu' +
+                  kTestValues2['flutter.String']),
+          throwsA(isA<PlatformException>()));
+      await preferences.reload();
```

Comment:
They're just a safety measure to make sure that no values in the dart-side cache get in the way, e.g. by tests that might get added in the future. Values that fail to be stored on the java side are still available in the dart-side cache (which is for another PR to resolve because it might break existing behavior).

Reply:
> Values that fail to be stored on the java side are still available in the dart-side cache (which is for another PR to resolve because it might break existing behavior).

That doesn't sound great; could you file an issue with details?